sysboundary/ionosphere.cpp

```diff
@@ -2222,11 +2230,18 @@ namespace SBC {
        }
 
        iSolverReal residualnorm = 0;
-       #pragma omp parallel for reduction(+:residualnorm)
+       iSolverReal maxAbsResidual = 0;
+       iSolverReal maxRelResidual = 0;
+       #pragma omp parallel for reduction(+:residualnorm) reduction(max:maxAbsResidual,maxRelResidual)
        for(uint n=0; n<nodes.size(); n++) {
          Node& N=nodes[n];
-         if(gaugeFixing == Equator && fabs(N.x[2]) < Ionosphere::innerRadius / 10.) {
+//if (rank == 0) {
+//   cerr << __FILE__ << ":" << __LINE__ << " gauge Equator " << ionosphereGrid.gaugeFixing << " " << Equator << " fabs(N.x[2]) " << fabs(N.x[2]) << " r sin " << Ionosphere::innerRadius * sin(Ionosphere::shieldingLatitude * M_PI / 180.0) << endl;
+//}
+//         if(true && fabs(N.x[2]) < Ionosphere::innerRadius * sin(Ionosphere::shieldingLatitude * M_PI / 180.0)) {
+         if(N.xMapped[0] == 0 && N.xMapped[1] == 0 && N.xMapped[2] == 0) {
             // Skip nodes that are not participating in the calculation.
+            N.parameters[ionosphereParameters::RESIDUAL] = 0;
```

Comment:
If there's anything to save from this PR, then it's this line. The rest I'm not so sure about any more.